### DISCUSSION TITLE
Feature Request: New Text object for mini.indentscope

### DISCUSSION THREAD

**Discussion by kevintraver** [nvim-mini/mini.nvim#453] [Ideas]
It would be useful to have a new text object that is similar to the existing text objects, but a new version that is restricted or bounded by newlines at the same indentation level:

Example:
```
    def sample_function():
        print("hello world")

        print("some text")
        print("more *text")      <--- cursor here
        print("even more text")

        print("some more text")
```
This would select:
```
        print("some text")
        print("more text")
        print("even more text")
```

**Comment by echasnovski** [comment]
Thanks for suggestion!

I don't think it is a very good idea. This is very close to `ip` text object which (not 100%, though, I understand). I think the current 'mini.indentscope' functionality is good enough.

**Reply by kevintraver** [reply]
Ok. Maybe I'll try to create my own new textobject.

Here's why `ap` and `ip` don't work:

```
    def sample_function()
        print("hello world")

        print("some text")
        print("more text")
        print("even more text")

        print("some more *text")      <--- cursor here
    def another_function()
        print "random text"
```

`ip`/`ap` will select this:
```
        print("some more text")
    def another_function()
        print "random text"
````

**Reply by echasnovski** [reply]
> Here's why `ap` and `ip` don't work:
> 
> ```
>     def sample_function()
>         print("hello world")
> 
>         print("some text")
>         print("more text")
>         print("even more text")
> 
>         print("some more *text")      <--- cursor here
>     def another_function()
>         print "random text"
> ```

I would do `V{j` in this case. And `V}k` in mirrored one. `ii` and `ip` should cover other cases.
